hivemq-edge/src/frontend/src/components/rjsf/BatchSubscription/components/ColumnMatcherStep.tsx

```diff
@@ -1,7 +1,128 @@
-import { FC } from 'react'
+import { FC, useEffect, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { JSONSchema7 } from 'json-schema'
+import { Select } from 'chakra-react-select'
+import { Controller, useFieldArray, useForm } from 'react-hook-form'
+import { Box, HStack, Input, chakra, FormControl, FormErrorMessage } from '@chakra-ui/react'
+import { LuChevronsRight } from 'react-icons/lu'
 
-const ColumnMatcherStep: FC = () => {
-  return <div></div>
+import { ColumnMappingData, ColumnOption, StepRendererProps } from '@/components/rjsf/BatchSubscription/types.ts'
+import { findMatch } from '@/components/rjsf/BatchSubscription/utils/levenshtein.utils.ts'
+
+const ColumnMatcherStep: FC<StepRendererProps> = ({ store, onContinue }) => {
+  const { schema, worksheet } = store
+  const { t } = useTranslation('components')
+
+  const subscriptions = useMemo<ColumnOption[]>(() => {
+    const { required, properties } = schema.items as JSONSchema7
+
+    //  TODO[NVL] If not required properties, do we "force" some (assuming a mqtt-topic for example)
+    return required
+      ? required.map((e) => {
+          const sss = properties?.[e] as JSONSchema7 | undefined
+          return { value: e, label: sss?.title || e }
+        })
+      : []
+  }, [schema.items])
```

Comment:
suggestion: I spotted some variables that might require some love in the longrun

```suggestion
    //  TODO[NVL] If not required properties, do we "force" some (assuming a mqtt-topic for example)
    return required
      ? required.map((requiredItem) => {
          const requiredItemProperties = properties?.[requiredItem] as JSONSchema7 | undefined
          return { value: requiredItem, label: requiredItemProperties?.title || requiredItem }
        })
      : []
  }, [schema.items])
```